pkg/manager/internal/controllers/organizationroletemplate_controller.go

```diff
@@ -53,6 +64,31 @@ func (r *OrganizationRoleTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	organizations := &corev1alpha1.OrganizationList{}
+	if err := r.Client.List(ctx, organizations); err != nil {
+		return ctrl.Result{}, fmt.Errorf("listing Organizations: %w", err)
+	}
+
+	var targets []corev1alpha1.OrganizationRoleTemplateTarget
+	for _, organization := range organizations.Items {
+		if organization.Status.Namespace != nil && organization.Status.Namespace.Name != "" {
```

Comment:
Check if the organization is ready; don't check specific details. If the organization is ready, those namespaces should exist. 

(This also enables future upgrades, what if the organization has to do further things before being ready, ones which the organizationRoleTemplate controller depends on. That way you aren't explicitly coding here what means to be ready enough for OrganizationRoleTemplate doing its job. Alternatively, you might check specific Organization conditions, but not check specific status fields. )

Reply:
You are absolutely right, I don't really remember why I didn't use the `Ready` function for this single case, maybe due to coding at night :)